playwright/pages/addressesListPage.ts

```diff
@@ -0,0 +1,106 @@
+    readonly setAsDefaultShipping = page.getByTestId("set-default-shipping-address"),
+    readonly deleteDialog = page.getByTestId("delete-address-dialog-content")
+  ) {
+    super(page);
+  }
+  async clickShowMoreMenu(addressPart: string) {
+    await this.addressCard
+      .filter({ hasText: addressPart })
+      .locator(this.showMoreMenuButton)
+      .click();
+  }
+  async clickManageAddresses() {
+    await this.menageAddressButton.click();
+  }
+  async clickAddAddressButton(){
+    await this.addAddressButton.click()
+  }
+  async clickEditAddress() {
+    await this.editAddressButton.click();
+  }
+    async clickDeleteAddress() {
+    await this.deleteAddressButton.click();
+  }
+  async setAsDeafultShippingAddress() {
+    await this.setAsDefaultShipping.click();
+  }
+  async setAsDeafultBillingAddress() {
+    await this.setAsDefaultBilling.click();
+  }
+  async verifyAddressFields(
+    addressPart: string,
+    address: AddressFieldsType,
+    matching: boolean = true,
+  ) {
+    const addressToVerify = await this.savedAddress.filter({ hasText: addressPart });
+    if (addressToVerify) {
+    const city = address.city.toUpperCase();
+    if (matching == true) {
+      expect(addressToVerify.getByTestId("name")).toContainText(
+        `${address.firstName} ${address.lastName}`,
+      );
+      if (address.companyName) {
+        expect(addressToVerify.getByTestId("company-name")).toContainText(address.companyName);
+      } else {
+        expect(this.companyName).toBeEmpty();
+      }
+      if (address.phone) {
+        expect(addressToVerify.getByTestId("phone")).toContainText(address.phone);
+      } else {
+        expect(this.phone).toBeEmpty();
+      }
+      expect(addressToVerify.getByTestId("addressLines")).toContainText(address.addressLine1);
+      if (address.addressLine2) {
+        expect(addressToVerify.getByTestId("addressLines")).toContainText(address.addressLine2);
+      }
+      expect(addressToVerify.getByTestId("postal-code-and-city")).toContainText(` ${address.zip} ${city}`);
+      expect(addressToVerify.getByTestId("country-area-and-country")).toContainText(address.country);
+    } else {
+      expect(addressToVerify.getByTestId("name")).not.toContainText(
+        `${address.firstName} ${address.lastName}`,
+      );
+      if (address.companyName) {
+        expect(addressToVerify.getByTestId("company-name")).not.toContainText(address.companyName);
+      } else {
+        expect(this.companyName).toBeEmpty();
+      }
+      if (address.phone) {
+        expect(addressToVerify.getByTestId("phone")).not.toContainText(address.phone);
+      } else {
+        expect(this.phone).toBeEmpty();
+      }
+      expect(addressToVerify.getByTestId("addressLines")).not.toContainText(address.addressLine1);
+      if (address.addressLine2) {
+        expect(addressToVerify.getByTestId("addressLines")).not.toContainText(address.addressLine2);
+      }
```

Comment:
I think we should not be using conditional statements this way. It is not readable and doing any change will be very painful in short time.

Reply:
@wojteknowacki fixed https://github.com/saleor/saleor-dashboard/pull/4657/commits/0560f72238fb42e11de6b96d91f1570cbb37fe2d